_posts/blog/2016-12-19-running-on-fedora-25.md

```diff
@@ -0,0 +1,335 @@
+
+Now that Accumulo is initialized, you can start its services:
+
+```bash
+sudo systemctl start accumulo-{master,tserver,gc,tracer,monitor}.service
+```
+
+Enable the commands to start at boot:
+
+```bash
+sudo systemctl enable accumulo-{master,tserver,gc,tracer,monitor}.service
+```
+
+## Running multiple nodes
+
+### Amazon EC2 setup
+
+For a multi-node setup, we used a Fedora 25 Cloud AMI on Amazon EC2, HVM
+virtualization type, with gp2 disk type with the root volume on EBS. Launched
+using 64GB for the root volume and did not attach additional storage. The
+latest AMI, as of this writing, seems to be `ami-e5757bf2` (`us-east-1`
+availability zone). For this example, launch three instances (in the same
+security group). The default user is `fedora` for the Fedora AMIs. Don't forget
+to make sure they are fully up-to-date (`sudo dnf --refresh upgrade`).
+
+For this setup, you should have a name service configured properly. For
+convenience, we used the EC2 provided internal DNS, with internal IP addresses.
+Make sure the nodes can communicate with each other using these names. If
+you're using EC2, this means making sure they are in the same security group,
+and the security group has an inbound rule for "All traffic" with the source
+set to the security group (`sg-xxxxxxxx`).
+
+### Configure and run Hadoop
+
+Set up Hadoop HDFS. This is the primary place we differ from the single node
+setup. For both Hadoop and Accumulo, you can edit the configuration files on
+one machine, and copy them to the others.
+
+Pick a server to be the Hadoop NameNode and identify its DNS name,
+(`<namenode-dns-name>`). Edit Hadoop's configuration to set the default
+filesystem name to this location:
+
+```bash
+sudo vim /etc/hadoop/core-site.xml
+```
+
+Set the value for the property `fs.default.name` to
+`hdfs://<namenode-dns-name>:8020`.
+
+Distribute copies of the changed configuration files to each node.
+
+Now, format the NameNode. You'll need to do this as the `hdfs` user on the
+NameNode instance:
+
+```bash
+sudo -u hdfs hdfs namenode -format
+```
+
+On the NameNode, start the NameNode service and enable it on reboot:
+
+```bash
+sudo systemctl start hadoop-namenode.service
+sudo systemctl enable hadoop-namenode.service
+```
+
+On each DataNode, start the DataNode service:
+
+```bash
+sudo systemctl start hadoop-datanode.service
+sudo systemctl enable hadoop-datanode.service
+```
+
+### Configure and run Accumulo
+
+Update Accumulo's configuration to use this HDFS filesystem:
+
+```bash
+sudo vim /etc/accumulo/accumulo-site.xml
+```
+
+Change the value of the `instance.volumes` to
+`hdfs://<namenode-dns-name>:8020/accumulo` in the configuration file. Don't
+forget to also change the default `instance.secret` and the trace user's
+credentials, if necessary. Also, since we'll have multiple nodes, you cannot
+use `localhost:2181` for ZooKeeper, so set `instance.zookeeper.host` to
+`<namenode-dns-name>:2181`. If you're running ZooKeeper on one of the other
```

Comment:
I think defining a convention of `<zk-server>` for the entire document would make things less wordy here and make the entire document easier to understand.